mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -235,6 +251,118 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
+    // 2. merge (gemmG, gemmKSplit) -> (gemmG*)
+
+    StringRef preservedDimName;
+    for (auto &dimName : gemmOperand.inputDimNames) {
+      if ((dimName != "gemmK") && (dimName != "gemmG"))
+        preservedDimName = dimName;
+    }
+
+    llvm::StringMap<uint32_t> unmergeDimOutputNames = expandNamesInPlace(
+        gemmOperand.inputDimNames, {{"gemmK", {"gemmKSplit", "gemmK"}}});
+
+    BottomUpTMBuilder unmergeTransform(builder, gemmOperand.inputDimNames,
+                                       gemmOperand.inputShape, loc);
+
+    BottomUpTMTopDimsWrapper unmergeTransformWrapper(
+        unmergeTransform, std::move(unmergeDimOutputNames));
+
+    unmergeTransformWrapper.passThrough("gemmG");
+    unmergeTransformWrapper.unmerge({"gemmKSplit", "gemmK"}, "gemmK",
+                                    {splitKFactor, K / splitKFactor});
+    unmergeTransformWrapper.passThrough(preservedDimName);
+
+    auto unmergeTransformAttr = unmergeTransform.get();
+    Value unmerge =
+        builder.create<TransformOp>(loc, gemmOperand.in, unmergeTransformAttr);
+
+    auto mergeTransform =
+        BottomUpTMBuilder::above(unmergeTransform, unmergeTransformAttr);
+
+    llvm::StringMap<uint32_t> mergeDimOutputNames =
+        expandNamesInPlace(gemmOperand.inputDimNames, {});
+
+    BottomUpTMTopDimsWrapper mergeTransformWrapper(
+        mergeTransform, std::move(mergeDimOutputNames));
+
+    mergeTransformWrapper.merge("gemmG", {"gemmG", "gemmKSplit"});
+    mergeTransformWrapper.passThrough(preservedDimName);
+    mergeTransformWrapper.passThrough("gemmK");
+
+    auto mergeTransformAttr = mergeTransform.get();
+    gemmOperand.out =
+        builder.create<TransformOp>(loc, unmerge, mergeTransformAttr);
+  }
+
+  {
+    // Prepare matrix C - i.e., (gemmG, gemmM, gemmN)
+    // 1. embed (gemmG) -> (~gemmG, gemmKSplit)[1, 0]
+    // 2. merge (~gemmG, gemmKSplit) -> (gemmG*)
+
+    SmallVector<StringRef> inputDimNames{"gemmG", "gemmM", "gemmN"};
+    BottomUpTMBuilder embedTransform(builder, inputDimNames, cShape, loc);
+
+    llvm::StringMap<uint32_t> embedDims = expandNamesInPlace(
+        inputDimNames, {{"gemmG", {"gemmGTilda", "gemmKSplit"}}});
+
+    const int64_t G = cShape[0];
+
+    BottomUpTMTopDimsWrapper embedWrap(embedTransform, std::move(embedDims));
+    embedWrap.embed({"gemmGTilda", "gemmKSplit"}, {G, splitKFactor}, "gemmG",
+                    {1, 0});
+    embedWrap.passThrough({"gemmM", "gemmN"});
+
+    auto embedTransformAttr = embedTransform.get();
+    Value embed = builder.create<TransformOp>(loc, c, embedTransformAttr);
+
+    auto mergeTransform =
+        BottomUpTMBuilder::above(embedTransform, embedTransformAttr);
+
+    llvm::StringMap<uint32_t> mergeDimOutputNames =
+        expandNamesInPlace(inputDimNames, {});
+
+    BottomUpTMTopDimsWrapper mergeTransformWrapper(
+        mergeTransform, std::move(mergeDimOutputNames));
+
+    mergeTransformWrapper.merge("gemmG", {"gemmGTilda", "gemmKSplit"});
```

Comment:
This can just be a passthrough : "gemmGTilda" --> "gemmG"
alongside : ignore("gemmKSplit").